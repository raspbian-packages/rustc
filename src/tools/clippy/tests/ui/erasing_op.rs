


#[allow(no_effect)]
#[warn(erasing_op)]
fn main() {
    let x: u8 = 0;

    x * 0;
    0 & x;
    0 / x;
}
