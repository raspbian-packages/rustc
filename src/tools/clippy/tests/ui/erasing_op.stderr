error: this operation will always return zero. This is likely not the intended outcome
 --> $DIR/erasing_op.rs:9:5
  |
9 |     x * 0;
  |     ^^^^^
  |
  = note: `-D erasing-op` implied by `-D warnings`

error: this operation will always return zero. This is likely not the intended outcome
  --> $DIR/erasing_op.rs:10:5
   |
10 |     0 & x;
   |     ^^^^^

error: this operation will always return zero. This is likely not the intended outcome
  --> $DIR/erasing_op.rs:11:5
   |
11 |     0 / x;
   |     ^^^^^

error: aborting due to 3 previous errors

